robot/web/stream/server.go

```diff
@@ -177,66 +177,29 @@ func (server *Server) AddStream(ctx context.Context, req *streampb.AddStreamRequ
 		return nil, errors.Errorf("stream is neither a camera nor audioinput. streamName: %v", streamStateToAdd.Stream)
 	}
 
+	var nameToPeerState map[string]*peerState
 	// return error if the caller's peer connection is already being sent stream data
-	if _, ok := server.activePeerStreams[pc][req.Name]; ok {
-		err := errors.New("stream already active")
-		server.logger.Error(err.Error())
-		return nil, err
-	}
-	nameToPeerState, ok := server.activePeerStreams[pc]
-	// if there is no active video data being sent, set up a callback to remove the peer connection from
-	// the active streams & stop the stream from doing h264 encode if this is the last peer connection
-	// subcribed to the camera's video feed
-	// the callback fires when the peer connection state changes & runs the cleanup routine when the
-	// peer connection is in a terminal state.
-	if !ok {
+	if pcStreams, pcHasStreams := server.activePeerStreams[pc]; pcHasStreams {
+		nameToPeerState = pcStreams
+		if _, isStreaming := pcStreams[req.Name]; isStreaming {
+			err := errors.New("stream already active")
+			server.logger.Error(err.Error())
+			return nil, err
+		}
+	} else {
+		// if there is no active video data being sent, set up a callback to remove the peer
+		// connection from the active streams & stop the stream from doing h264 encode if this is
+		// the last peer connection subcribed to the camera's video feed the callback fires when the
+		// peer connection state changes & runs the cleanup routine when the peer connection is in a
+		// terminal state.
 		nameToPeerState = map[string]*peerState{}
+		server.activePeerStreams[pc] = nameToPeerState
 		pc.OnConnectionStateChange(func(peerConnectionState webrtc.PeerConnectionState) {
-			server.logger.Infof("%s pc.OnConnectionStateChange state: %s", req.Name, peerConnectionState)
```

Comment:
I just gave this function a name. Now that I'm scrutinizing this the state of `activePeerStreams`, I found this big blob (in part due to its large comment) a distraction. Notably [this assignment](https://github.com/viamrobotics/rdk/pull/4932/files#diff-b664e7f5f24e04be1ed075bf7d0f4209eaf9fc88f34eb7acc987ad1693857223L239) that could be happening before constructing the callback.